### ISSUE TITLE
Only 62% accuracy.How to make it better?

### ISSUE THREAD

**Issue by ll36771** [bgshih/crnn#5]
I train a new model and use crnn_demo_model.t7 as a pretrained model.Now iteration is 30000,train loss is 0.0005,but test loss is 4.6.And accuracy is around 62% for a long time.
Is anything wrong with it?
What can i do to make it better?
@bgshih

**Comment by bgshih** [comment]
Looks like your model is overfitting. How much data did you use?

**Comment by ll36771** [comment]
Thanks for your reply.I use around 20000 samples.I know it's not too much.I just want to have a test.Should i add dropout?
@bgshih

**Comment by bgshih** [comment]
@ll36771  I think the number of samples is adequate for fine-tuning. Are you using lower learning rates at RNNs? Dropout may also help, but I haven't tried yet.

**Comment by ll36771** [comment]
No,i use the deafult setting of optim till now.I only change the batchsize to 64.Where to change the rnn learning rates?

**Comment by ll36771** [comment]
@bgshih

**Comment by ll36771** [comment]
@bgshih

**Comment by bgshih** [comment]
@ll36771 I don't know an easy way to set different learning rates for different layers. Setting `config.learningRates` works, but needs some hacks.

**Comment by moraval** [comment]
Hi @bgshih,

I wanted to ask you, if this crnn_demo_model.t7 is the final model you used to get the results from your paper (http://arxiv.org/pdf/1507.05717v1.pdf). I tried to reproduce your results on Robust reading competition dataset and I got the accuracy of only 79.8% instead of your 86.7%. I wonder what could be the problem.

Thanks in advance for your reply.

**Comment by bgshih** [comment]
@moraval Thanks for your feedback. The model is different from the one we used in the paper. However, as I tested on IIIT, IC03, SVT, this one produces higher or very close results. The 79.8% accuracy on IC13 does not seem normal, and I am looking into this.
By the way, how many test samples did you use?

**Comment by moraval** [comment]
After downloading the name of the file is: "Challenge2_Test_Task3_Images" and there were 1095 images. The first word is "Tiredness".

**Comment by bgshih** [comment]
@moraval The test set we use has 1015 samples, we only use samples that contain only alphanumeric characters and at least three characters, following the previous work http://arxiv.org/abs/1412.1842 , http://arxiv.org/abs/1412.5903, etc.

**Comment by bgshih** [comment]
@moraval However, I don't think this is the only reason.

**Comment by bgshih** [comment]
@moraval  Have you tested the model on other dataset, like the SVT?

**Comment by moraval** [comment]
Thanks for the answer. I'll create a dataset according to you and see if it improves.
@bgshih

**Comment by moraval** [comment]
@bgshih No, not yet. I first wanted to know, it I even should be concerned about not getting the correct accuracy.

**Comment by bgshih** [comment]
@moraval I was wrong about the number 1015. After removing the words that contain non-alphanumeric characters or less than 3 characters, there are 857 examples.

**Comment by moraval** [comment]
@bgshih I tested the model on the "selected" dataset and got the accuracy of: 0.885648 on 857 samples. I'll later do the testing on the other datasets, too. Thanks a lot for your help.

**Comment by ahmedmazari-dhatim** [comment]
Hi @bgshih @ll36771,

1- What is the appropriate number of samples needed for fine-tuning the model and the number of examples in each sample ?
2- What is the number of examples in each sample in the original trained model ?

Every comment would be appreciated.

Thank you
